CONTRIBUTING.md

```diff
@@ -0,0 +1,261 @@
+| 5. Run checks | `make lint && make test && make testacc` | Verify quality |
+| 6. Commit | `git commit -m "description"` | Save changes |
+| 7. Push | `git push origin feature-name` | Share work |
+| 8. Open PR | Create pull request on GitHub | Request review |
+
+## Branch Strategy
+
+| Branch | Purpose | When to Use |
+|--------|---------|-------------|
+| `main` | Stable releases | Bug fixes, stable features |
+| `dev-preview` | Preview features | Experimental features, breaking changes |
+
+**Default:** Target `main` for most contributions. Use `dev-preview` only for experimental features that need early feedback.
+
+## Requirements
+
+| Tool | Minimum Version | Installation Guide |
+|------|----------------|-------------------|
+| Go | Latest stable | https://go.dev/doc/install |
+| Terraform | 1.0+ | https://developer.hashicorp.com/terraform/install |
+| golangci-lint | Latest | https://golangci-lint.run/usage/install/ |
+| AAP/AWX instance | 2.4+ | https://github.com/ansible/awx/blob/devel/INSTALL.md |
+
+## Development Setup
+
+| Action | Command |
+|--------|---------|
+| Build provider | `make build` |
+| Configure local dev | See [Development Overrides](#local-testing) |
+
+### Local Testing
+
+Create a Terraform config file (`~/.terraformrc` or `$TF_CLI_CONFIG_FILE`):
+
+```hcl
+provider_installation {
+  dev_overrides {
+    "ansible/aap" = "/path/to/terraform-provider-aap"
+  }
+  direct {}
+}
+```
+
+## Testing Requirements
+
+### Unit Tests
+
+| Action | Command | Required? |
+|--------|---------|-----------|
+| Run all unit tests | `make test` | ✅ Yes |
+| Run with coverage | `make testcov` | ✅ Yes |
+| CI must pass | Automatic | ✅ Yes |
+
+**Requirements**: 
+- All PRs must include unit tests for new code
+- New code must have minimum 80% test coverage ([SonarCloud](https://sonarcloud.io/project/overview?id=ansible_terraform-provider-aap))
+
+Run `make testcov` to generate coverage report locally at `./unit-testing.cov`.
+
+### Acceptance Tests
+
+| Action | Command | Required? |
+|--------|---------|-----------|
+| Setup test resources | `ansible-playbook testing/playbook.yml` | First time only |
+| Source test vars | `source testing/acceptance_test_vars.env` | Before each run |
+| Run acceptance tests | `make testacc` | ✅ Yes |
+
+**Requirement**: PRs must include acceptance test output in PR description.
+
+#### Environment Setup
+
+```bash
+# Set AAP credentials
+export AAP_USERNAME=admin
+export AAP_PASSWORD=your_password
+export AAP_HOSTNAME=http://localhost:9080
+export AAP_INSECURE_SKIP_VERIFY=true
+export AAP_VALIDATE_CERTS=false
+
+# For AAP 2.4
+export CONTROLLER_OPTIONAL_API_URLPATTERN_PREFIX="/api/"
+
+# Install ansible collection
+ansible-galaxy collection install -r testing/requirements.yml
+
+# Create test resources
+ansible-playbook testing/playbook.yml
+```
+
+⚠️ **WARNING**: Acceptance tests modify real AAP resources. Use a dedicated test instance.
```

Comment:
This content also exists in the README. Instead of duplicating it (and having a second place to maintain it), can we reference the existing steps?